python/cudf/bindings/binops.pyx

```diff
@@ -161,30 +143,26 @@ def apply_op(lhs, rhs, out, op):
     free(c_out)
 
     return nullct
-  
-def apply_op_udf(lhs, rhs, out, ptx):
-    """
-    Dispatches a binary op call to the appropriate libcudf function:
-    """
-    check_gdf_compatibility(out)
-    cdef gdf_column* c_lhs = NULL
-    cdef gdf_column* c_rhs = NULL
-    cdef gdf_column* c_out = column_view_from_column(out)
-
+ 
+def apply_op_udf(lhs, rhs, udf_ptx, np_dtype):
+    
     check_gdf_compatibility(lhs)
     check_gdf_compatibility(rhs)
-    c_lhs = column_view_from_column(lhs)
-    c_rhs = column_view_from_column(rhs)
+    cdef gdf_column* c_lhs = column_view_from_column(lhs)
+    cdef gdf_column* c_rhs = column_view_from_column(rhs)
 
-    nullct = apply_op_v_v_udf(
-        <gdf_column*>c_lhs,
-        <gdf_column*>c_rhs,
-        <gdf_column*>c_out,
-        ptx
-    )
+    cdef gdf_column c_outcol
 
+    cdef gdf_dtype g_type = dtypes[np_dtype] # get the gdf_type related to the input np type 
+    
+    cdef string cpp_str = udf_ptx.encode('UTF-8')
+    
+    with nogil:
+        c_outcol = binary_operation(<gdf_column>c_lhs[0], <gdf_column>c_rhs[0], cpp_str, g_type)
```

Comment:
`c_outcol` is stack allocated but the device memory pointed from it is dynamically allocated. If this sentence is in `C++` it will be free automatically when going out of scope but I am not sure if this is the case for cython/python.